clusterversion: move to 25.4 version

annrpom opened the pull request:
**Checklist**:

- [x] Add version key constant for new release (e.g. `V24_2`), equal to `Latest`. Note that we do NOT update the `PreviousRelease` constant. We can only update the `PreviousRelease` when an RC is published, which is needed for upgrade roachtests.
- [x] Add final version for the previous release (e.g. `V24_1`).
- [x] Add start version (e.g. `V24_2Start` with version `24.1-2`) and add a new first upgrade for it (in `upgrades/upgrades.go`).
- [x] Update `roachpb.successorSeries` map and update `TestReleaseSeriesSuccessor`
- [x] Update `pkg/build/version.txt` to the new version (e.g. `v24.2.0-alpha.00000000`)
- [x] Update the `scplan` rules in `pkg/sql/schemachanger/scplan/internal/rules`:
  - copy the contents of `current` into a new release directory for the previous version (e.g. `release_24_1`)
  - change package name in all files (including `BUILD.bazel`), and update the version in `current/helpers.go`
  - update `rulesForReleases` in `scplan/plan.go`
  - rewrite the test outputs: `./dev test pkg/sql/schemachanger/scplan/internal/rules/... --rewrite`
  - rewrite `TestDeclarativeRules` output: `./dev test pkg/cli -f DeclarativeRules --rewrite`
- [x] Update releases file: ``` bazel build //pkg/cmd/release:release _bazel/bin/pkg/cmd/release/release_/release update-releases-file ``` Note that the forked release version will disappear from the releases file. This is what we want; if it is here, we will attempt to run upgrade tests against it.
- [x] Regenerate expected test data as needed (usually `pkg/sql/catalog/systemschema_test` and some logictests).

Epic: REL-2733
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/149494)
<!-- Reviewable:end -->

annrpom committed:
clusterversion: move to 25.4 version

**Checklist**:

- [x] Add version key constant for new release (e.g. `V24_2`), equal to `Latest`.
  Note that we do NOT update the `PreviousRelease` constant. We can only update the
  `PreviousRelease` when an RC is published, which is needed for upgrade
  roachtests.
- [x] Add final version for the previous release (e.g. `V24_1`).
- [x] Add start version (e.g. `V24_2Start` with version `24.1-2`) and add a new
  first upgrade for it (in `upgrades/upgrades.go`).
- [x] Update `roachpb.successorSeries` map and update `TestReleaseSeriesSuccessor`
- [x] Update `pkg/build/version.txt` to the new version (e.g. `v24.2.0-alpha.00000000`)
- [x] Update releases file:
  ```
  bazel build //pkg/cmd/release:release
  _bazel/bin/pkg/cmd/release/release_/release update-releases-file
  ```
  Note that the forked release version will disappear from the releases file. This
  is what we want; if it is here, we will attempt to run upgrade tests against it.
- [x] Regenerate expected test data as needed (usually
  `pkg/sql/catalog/systemschema_test` and some logictests).

Epic: REL-2733
Release note: None

annrpom committed:
schemachanger: generate deprules for 25.4

**Checklist**:

- [x] Update the `scplan` rules in `pkg/sql/schemachanger/scplan/internal/rules`:
  - copy the contents of `current` into a new release directory for the previous version
    (e.g. `release_24_1`)
  - change package name in all files (including `BUILD.bazel`), and update the
    version in `current/helpers.go`
  - update `rulesForReleases` in `scplan/plan.go`
  - rewrite the test outputs: `./dev test pkg/sql/schemachanger/scplan/internal/rules/... --rewrite`
  - rewrite `TestDeclarativeRules` output: `./dev test pkg/cli -f DeclarativeRules --rewrite`

Epic: REL-2733
Release note: None

celiala commented on pkg/clusterversion/cockroach_versions.go:
```diff
@@ -279,6 +286,10 @@ var versionTable = [numKeys]roachpb.Version{
 
+	V25_3: {Major: 25, Minor: 3, Internal: 0},
+
+	// v25.4 versions. Internal versions must be even.
```
good catch - thanks for adding this note (and backfilling the one for 25.3)

celiala reviewed (approved):
this LGTM - thank you!!

annrpom commented:
TFTR! ('-')7

bors r+
